src/cljdoc/render/api_searchset.clj

```diff
@@ -29,27 +27,119 @@
   [version-entity name-of-ns def-name]
   (str (path-for-namespace version-entity name-of-ns) "#" def-name))
 
-(defn- elements->doc-segments
-  "Does the tricky work of walking through the Jsoup node siblings and collecting
-  headers and their text. Important considerations:
+(defn- soup-elem?
+  "Returns true if Jsoup elem is an html element (as oppposed to a text node)"
+  [elem]
+  (instance? org.jsoup.nodes.Element elem))
 
-  1. Documents do not necessarily start with a header.
-  2. Headers do not necessarily have text between them and the next header or the
-     end of the document.
-  3. Flexmark makes sure that all markdown headers have anchors, meaning this only works for markdown and not asciidoc."
-  [elements doc-title]
+(defn heading-elem?
+  [elem]
+  (and
+   (soup-elem? elem)
+   (#{"h1" "h2" "h3" "h4" "h5" "h6"} (.tagName elem))
+   (= "a" (some-> elem .children first .tagName))))
+
+(defn- adoc-section? [elem]
+  (and (soup-elem? elem)
+       (= "div" (.tagName elem))
+       (.hasAttr elem "class")
+       (re-matches #"sect[1-9]" (.attr elem "class"))
+       (heading-elem? (first (.children elem)))))
+
+(defn- adoc-section-title [section-elem]
+  (-> section-elem .children .first .text))
+
+(defn- adoc-section-anchor [section-elem]
+  (-> section-elem .children .first .attributes (.get "id")))
+
+(defn- append-text [s elem]
+  (let [t (-> elem .text str/trim)]
+    (if (= "" t)
+      s
+      (str s t " "))))
```

Comment:
do we need a space between s and t here?

Reply:
Thanks, for checking, I'm not sure. 

Let's say we have t1 t2 and t3.

How would s would progress...
1. ""
2. "t1 "
3. "t1 t2 "
4. "t1 t2 t3 "

So, I don't think so! But... happy to be wrong!

Reply:
I _think_ we are fine... 
If I've got a blind spot we can fix later.

Reply:
ohh ok yes this does make sense, thanks!! 💜